### PR TITLE
Added all current available AWS regions

### PR DESCRIPTION
**Pull request by bolt12**
###### Description of changes

Added all currently available AWS regions to create-amis.sh file.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/maintainers/scripts/ec2/create-amis.sh [hunk 1]
```diff
@@ -26,12 +26,32 @@ var ${home_region:=eu-west-1}
 var ${bucket:=nixos-amis}
 var ${service_role_name:=vmimport}
 
-var ${regions:=eu-west-1 eu-west-2 eu-west-3 eu-central-1 eu-north-1
-         us-east-1 us-east-2 us-west-1 us-west-2
+# Output of the command:
+# > aws ec2 describe-regions --all-regions --query "Regions[].{Name:RegionName}" --output text
```
Comment (roberth) [inline nixos/maintainers/scripts/ec2/create-amis.sh:30]:
```suggestion
# > aws ec2 describe-regions --all-regions --query "Regions[].{Name:RegionName}" --output text | sort
```

I don't know what determines the current order, but we'll regret (slightly) not sorting if it changes.

**Review by roberth — COMMENTED** [review]
Seems like a good idea to sort.
Would be good to get this in soon, as it may be used at release time.

File: nixos/maintainers/scripts/ec2/create-amis.sh (hunk 1 above)
Comment (bolt12) [reply nixos/maintainers/scripts/ec2/create-amis.sh:30]: Great point!

Comment (bolt12) [reply nixos/maintainers/scripts/ec2/create-amis.sh]: Addressed!

**Review by roberth — COMMENTED** [review]
:+1:

**Review by roberth — APPROVED** [review]

**Comment by AmineChikhaoui** [comment]
Hi, iirc the regions need to be enabled through the AWS account first. Since the diff isn't very clear, can you list what regions where added in this PR ?

**Comment by bolt12** [comment]
@AmineChikhaoui you'd have to enable all regions since this PR adds all regions currently available. There's also something else that might be needed to do which has to do with this warning on the Enable Regions dashboard:

> AWS recommends using regional AWS Security Token Service (STS) endpoints to reduce latency. Session tokens from regional STS endpoints are valid in all AWS Regions. If you use regional STS endpoints, no action is required.
>
> Session tokens from the global STS endpoint (https://sts.amazonaws.com) are valid only in AWS Regions that are enabled by default. If you intend to enable a new Region for your account, you can either use session tokens from regional STS endpoints or activate the global STS endpoint to issue session tokens that are valid in all AWS Regions. You can do this in[ Account Settings](https://us-east-1.console.aws.amazon.com/iam/home#/account_settings) in the IAM console.
>
> Session tokens that are valid in all AWS Regions are larger. If you store session tokens, these larger tokens might affect your systems. [Learn more](https://docs.aws.amazon.com/console/iam/region-session-tokens-size)

So this might be need enabling as well

**Review by AmineChikhaoui — APPROVED** [review]

**Pull request merged**